setup.py

```diff
@@ -0,0 +1,83 @@
+from setuptools import setup, find_packages
+import sys
+import os
+
+# PEP0440 compatible formatted version, see:
+# https://www.python.org/dev/peps/pep-0440/
+#
+# release markers:
+#   X.Y
+#   X.Y.Z   # For bugfix releases
+#
+# pre-release markers:
+#   X.YaN   # Alpha release
+#   X.YbN   # Beta release
+#   X.YrcN  # Release Candidate
+#   X.Y     # Final release
+
+# version.py defines the VERSION and VERSION_SHORT variables.
+# We use exec here so we don't import allennlp_semparse whilst setting up.
+VERSION = {}
+with open("allennlp_rc/version.py") as version_file:
+    exec(version_file.read(), VERSION)
+
+# Load requirements.txt with a special case for allennlp so we can handle
+# cross-library integration testing.
+with open("requirements.txt") as requirements_file:
+    install_requirements = requirements_file.readlines()
+    install_requirements = [
+        r for r in install_requirements if "git+https://github.com/allenai/allennlp" not in r
+    ]
+    if not os.environ.get("EXCLUDE_ALLENNLP_IN_SETUP"):
+        # Warning: This will not give you the desired version if you've already
+        # installed allennlp! See https://github.com/pypa/pip/issues/5898.
+        #
+        # There used to be an alternative to this using `dependency_links`
+        # (https://stackoverflow.com/questions/3472430), but pip decided to
+        # remove this in version 19 breaking numerous projects in the process.
+        # See https://github.com/pypa/pip/issues/6162.
+        #
+        # As a mitigation, run `pip uninstall allennlp` before installing this
+        # package.
+        sha = "8bc5da2d7d0050374c78b52adde7149425e3b52a"
```

Comment:
This should match the line in `requirements.txt`, IIRC. The redundancy is annoying, but the syntax isn't the same in `requirements.txt` and here. Grabbing the version with a regex would maybe be cleaner, but also seemed liable to break.